### ISSUE TITLE
ExecutionOrder.cs robustness issues

### ISSUE THREAD

**Issue by nickjbenson** [leapmotion/UnityModules#494]
The [ExecuteBefore], [ExecuteAfter], and [ExecuteBeforeDefault]/[ExecuteAfterDefault] attributes were super handy to set up easily-readable (and flexible!) execution order constraints on-the-go while building parts of the UI module revamp, but after a number of compilations (and checkouts to and from other branches...) it started complaining, "Not enough execution indexes to fit all of the scripts!" This shouldn't be happening based solely on the attributes themselves; the doc comments' explanation of that error is definitely not what's going on.

Assigning the author, Amarcolina. These attributes are pretty handy. It would be great if they were reliable... But if it's not worth it to fix it, we should remove them so users don't stumble upon them and hit upon these same issues.

**Comment by Amarcolina** [comment]
Want to discuss removing this attribute altogether.  We never like relying on execution order and so I think we could switch to an entirely event based system and do away with execution order entirely????

**Comment by nickjbenson** [comment]
I am in support of removing the attribute in favor of event-based systems or manager systems.

In the Interaction Engine, we resolve execution order problems by having everything IE choreographed by the Interaction Manager. This is basically the same as having an Component-System configuration, where InteractionBehaviours are the Components updated by the InteractionManager system. Same deal with the Graphic Renderer component and Feature subcomponents. We'll probably use this pattern a lot.

The Hand Controller is a different story, since it can't really be a global manager for all hand-related things, so an event-based model makes more sense and is actually what it uses.

In other words, I don't think it's worth maintaining this feature, since we have good alternatives.
